From d56a986bc6fa52be3b007d9539d914877c845926 Mon Sep 17 00:00:00 2001
From: Dinesh Sudham <dineshx.sudham@intel.com>
Date: Tue, 9 Jun 2020 12:17:56 +0800
Subject: [PATCH] PONRTSYS-1804:XPCS: MPLLB settings for per speed

MPLLB frequency and bandwidth settings are changes per speed
When we change from one speed to another need to make sure
proper frequency and bandwidth selected for per speed.

Signed-off-by: Dinesh Sudham <dineshx.sudham@intel.com>
---
 drivers/net/ethernet/lantiq/xpcs/xpcs.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/xpcs/xpcs.c b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
index 1a3c4f07dbe7..46adc0b1ae4d 100644
--- a/drivers/net/ethernet/lantiq/xpcs/xpcs.c
+++ b/drivers/net/ethernet/lantiq/xpcs/xpcs.c
@@ -635,6 +635,8 @@ static int xpcs_teng_xaui_mode(struct xpcs_prv_data *pdata)
 {
 	struct xpcs_serdes_cfg *serdes_cfg = pdata->serdes_cfg;
 
+	pdata->mpllb = 0;
+
 	if (xpcs_byp_pwrupseq(pdata) != XPCS_SUCCESS)
 		return XPCS_FAILURE;
 
@@ -706,6 +708,8 @@ static int xpcs_teng_kr_mode(struct xpcs_prv_data *pdata)
 {
 	struct xpcs_serdes_cfg *serdes_cfg = pdata->serdes_cfg;
 
+	pdata->mpllb = 0;
+
 	if (xpcs_byp_pwrupseq(pdata) != XPCS_SUCCESS)
 		return XPCS_FAILURE;
 
@@ -774,6 +778,7 @@ static int xpcs_oneg_xaui_mode(struct xpcs_prv_data *pdata)
 {
 	struct xpcs_serdes_cfg *serdes_cfg = pdata->serdes_cfg;
 
+	pdata->mpllb = 0;
 	pdata->sgmii_type = PHY_SIDE_SGMII;
 
 	if (xpcs_byp_pwrupseq(pdata) != XPCS_SUCCESS)
